src/Avalonia.Controls/AutoCompleteBox.cs

```diff
@@ -225,6 +225,27 @@ public enum AutoCompleteFilterMode
         Custom = 13,
     }
 
+    /// <summary>
+    /// Represents the selector used by the
+    /// <see cref="T:Avalonia.Controls.AutoCompleteBox" /> control to
+    /// determine how the specified text should be modified with an item.
+    /// </summary>
+    /// <returns>
+    /// Modified text that will be used by the
+    /// <see cref="T:Avalonia.Controls.AutoCompleteBox" />.
+    /// </returns>
+    /// <param name="search">The string used as the basis for filtering.</param>
+    /// <param name="item">
+    /// The selected item that should be combined with the
+    /// <paramref name="search" /> parameter.
+    /// </param>
+    /// <typeparam name="T">
+    /// The type used for filtering the
+    /// <see cref="T:Avalonia.Controls.AutoCompleteBox" />.
+    /// At the moment this type known only as a string.
```

Comment:
Probably we should provide selected item value in this delegate instead of formatted string.
Otherwise T doesn't have any sense for me.

Reply:
@maxkatz6, maybe we should do it by analogy with `TextFilter` and `ItemFilter`: `TextSelector` and `ItemSelector`. If the first is defined, a formatted value is passed to it, if not, we try to apply the `ItemSelector`, where the value itself is passed. If there is no `ItemSelector` either, we just replace the text with a formatted value. What do you think?

Reply:
Added `AutoCompleteSelector<object> ItemSelector` property to "mirror" `TextFilter/ItemFilter` pair. Now it makes sense :)

Reply:
Yes, it will work fine, but isn't it more complex or less clear?
I am also not sure if TextFilter providers any special information, that developer can't get - it uses either ToString or Binding value.

Reply:
@maxkatz, I dunno, tbh :) But `TextFilter` exists already for some reason. And I think the reason is that we can't provide truly generic filter, so either users deal with `object` all the time, or we provide them with an additional tool for the most common option (cause `string` is most often used for autocomplete)

Reply:
My bad, I meant TextSelector, not TextFilter.
But I see now, you kept both to make it consistent with existing API.

Reply:
@maxkatz6, that's the thing :)